### PR TITLE
[CIT-520] version name corner cases

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes
Code changes handle the corner cases with package version names containing "::".
Updated tests as well.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_serializer.rs [hunk 1]
```diff
@@ -241,6 +247,7 @@ mod tests {
     #[test]
     fn verify_combine_version_and_type() {
         let some_version: Option<SoftwareVersion> = Some("1.0".to_string());
+        let some_version_with_collon: Option<SoftwareVersion> = Some("1.0.0::1".to_string());
```
Comment (rina23q) [inline mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_serializer.rs:250]: Typo. `collon` => `colon`

File: mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_deserializer.rs [hunk 2]
```diff
@@ -122,15 +122,25 @@ impl SmartRestUpdateSoftware {
 impl SmartRestUpdateSoftwareModule {
     // this doesn't cover the corner case '1.0.0::dd::debian::' for now (escape)
```
Comment (rina23q) [inline mapper/tedge_mapper/src/sm_c8y_mapper/smartrest_deserializer.rs:123]: Can you remove this comment? You solved this :)

**Review by rina23q — APPROVED** [review]
Thank you! Just fix very minor comments, then merge it :)
